src/main/java/wallet/logic/parser/AddCommandParser.java

```diff
@@ -0,0 +1,208 @@
+            return null;
+        }
+        return null;
+    }
+
+    /**
+     * Returns an Expense object.
+     *
+     * @param input A string input.
+     * @return The Expense object.
+     * @throws NumberFormatException          Wrong format.
+     * @throws ArrayIndexOutOfBoundsException Out of index.
+     */
+    private Expense parseExpense(String input) throws NumberFormatException, ArrayIndexOutOfBoundsException {
+        boolean isRecurring = input.contains("/r");
+        String[] arguments = input.split("\\$");
+        String desc = arguments[0].trim();
+        arguments = arguments[1].split(" ", 2);
+        Double amount = Double.parseDouble(arguments[0].trim());
+        String cat;
+        LocalDate date = LocalDate.now();
+        DateTimeFormatter formatter = DateTimeFormatter.ofPattern("dd/MM/yyyy");
+        String freq = null;
+
+        if (arguments[1].contains("/on")) {
+            arguments = arguments[1].split(" ", 2);
+            cat = arguments[0].trim();
+            if (isRecurring) {
+                arguments = arguments[1].split("/on");
+                arguments = arguments[1].split("/r");
+                date = LocalDate.parse(arguments[0].trim(), formatter);
+                freq = arguments[1].trim().toUpperCase();
+                if (!freq.equals("DAILY") && !freq.equals("WEEKLY") && !freq.equals("MONTHLY")) {
+                    System.out.println(AddCommand.MESSAGE_ERROR_ADD_EXPENSE);
+                    System.out.println(AddCommand.MESSAGE_USAGE);
+                }
+            } else {
+                arguments = arguments[1].split("/on");
+                date = LocalDate.parse(arguments[1].trim(), formatter);
+            }
+        } else {
+            cat = arguments[1].trim();
+        }
+        Expense expense = new Expense(desc, date, amount, cat, isRecurring, freq);
+
+        return expense;
+    }
+
+    /**
+     * Returns a Loan object.
+     *
+     * @param input The string after "loan".
+     * @return The Loan object.
+     * @throws ArrayIndexOutOfBoundsException Out of index.
+     * @throws ParseException                 ParseException.
+     */
+    private Loan parseLoan(String input) throws ArrayIndexOutOfBoundsException, ParseException {
+        Loan loan = null;
+        Boolean isLend = false;
+
+        String[] info = input.split(" ", 4);
+        String description = info[0];
+        double amount = Double.parseDouble(info[1].replace("$", ""));
+        DateTimeFormatter formatter = DateTimeFormatter.ofPattern("dd/MM/yyyy");
+        LocalDate createdDate = LocalDate.parse(info[2].trim(), formatter);
+
+        info = info[3].split("/c ");
+        int contactId = Integer.parseInt(info[1]);
+
+        if (info[0].equals("/l")) {
+            isLend = true;
+        } else if (info[0].equals("/b")) {
+            isLend = false;
+        }
+
+        ArrayList<Contact> contactList = LogicManager.getWallet().getContactList().getContactList();
```

Comment:
.getContactList().getContactList()? Not just a naming issue but perhaps this can be done more efficiently with reduce dependencies.